app/controllers/services_controller.rb

```diff
@@ -0,0 +1,6 @@
+class ServicesController < ApplicationController
+  def register
+    @service = Service.create
```

Comment:
What happens if an invalid outside source tries to make a request? Should have some sort of internal verification (that uses a key/seed not visible on repo) to verify the key. This should also be usable by the engine so it can validate the requests coming into the engines (to update their database).